Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/CategorySample.cs

```diff
@@ -0,0 +1,258 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Runtime.InteropServices;
+
+    internal class CategorySample
+    {
+        public Dictionary<int, CounterDefinitionSample> CounterTable;
+
+        public Dictionary<string, int> InstanceNameTable;
+
+        private readonly PerfLib library;
+
+        public CategorySample(byte[] data, int categoryNameIndex, int counterNameIndex, PerfLib library)
```

Comment:
This is not our code, it's copied from CLR's PerformanceCounter implementation. It does certain amount of black magic regarding parsing a byte stream with complicated structure and cross-indexed fields. Unit testing this would defeat the purpose, since constructing the byte stream would be much more error-prone than the imiplementation itself. PerformanceCounter has been around for a while, so I'm taking it as is.